src/fsevents.c

```diff
@@ -0,0 +1,210 @@
+#define kFSEventStreamEventFlagUserDropped 0x00000002
+#endif
+
+#ifndef kFSEventStreamEventFlagKernelDropped
+#define kFSEventStreamEventFlagKernelDropped 0x00000004
+#endif
+
+#ifndef kFSEventStreamEventFlagEventIdsWrapped
+#define kFSEventStreamEventFlagEventIdsWrapped 0x00000008
+#endif
+
+#ifndef kFSEventStreamEventFlagHistoryDone
+#define kFSEventStreamEventFlagHistoryDone 0x00000010
+#endif
+
+#ifndef kFSEventStreamEventFlagRootChanged
+#define kFSEventStreamEventFlagRootChanged 0x00000020
+#endif
+
+#ifndef kFSEventStreamEventFlagMount
+#define kFSEventStreamEventFlagMount 0x00000040
+#endif
+
+#ifndef kFSEventStreamEventFlagUnmount
+#define kFSEventStreamEventFlagUnmount 0x00000080
+#endif
+
+#ifndef kFSEventStreamEventFlagItemCreated
+#define kFSEventStreamEventFlagItemCreated 0x00000100
+#endif
+
+#ifndef kFSEventStreamEventFlagItemRemoved
+#define kFSEventStreamEventFlagItemRemoved 0x00000200
+#endif
+
+#ifndef kFSEventStreamEventFlagItemInodeMetaMod
+#define kFSEventStreamEventFlagItemInodeMetaMod 0x00000400
+#endif
+
+#ifndef kFSEventStreamEventFlagItemRenamed
+#define kFSEventStreamEventFlagItemRenamed 0x00000800
+#endif
+
+#ifndef kFSEventStreamEventFlagItemModified
+#define kFSEventStreamEventFlagItemModified 0x00001000
+#endif
+
+#ifndef kFSEventStreamEventFlagItemFinderInfoMod
+#define kFSEventStreamEventFlagItemFinderInfoMod 0x00002000
+#endif
+
+#ifndef kFSEventStreamEventFlagItemChangeOwner
+#define kFSEventStreamEventFlagItemChangeOwner 0x00004000
+#endif
+
+#ifndef kFSEventStreamEventFlagItemXattrMod
+#define kFSEventStreamEventFlagItemXattrMod 0x00008000
+#endif
+
+#ifndef kFSEventStreamEventFlagItemIsFile
+#define kFSEventStreamEventFlagItemIsFile 0x00010000
+#endif
+
+#ifndef kFSEventStreamEventFlagItemIsDir
+#define kFSEventStreamEventFlagItemIsDir 0x00020000
+#endif
+
+#ifndef kFSEventStreamEventFlagItemIsSymlink
+#define kFSEventStreamEventFlagItemIsSymlink 0x00040000
+#endif
+
+// constants from https://developer.apple.com/library/mac/documentation/Darwin/Reference/FSEvents_Ref/index.html#//apple_ref/doc/constant_group/FSEventStreamCreateFlags
+#ifndef kFSEventStreamCreateFlagNone
+#define kFSEventStreamCreateFlagNone 0x00000000
+#endif
+
+#ifndef kFSEventStreamCreateFlagUseCFTypes
+#define kFSEventStreamCreateFlagUseCFTypes 0x00000001
+#endif
+
+#ifndef kFSEventStreamCreateFlagNoDefer
+#define kFSEventStreamCreateFlagNoDefer 0x00000002
+#endif
+
+#ifndef kFSEventStreamCreateFlagWatchRoot
+#define kFSEventStreamCreateFlagWatchRoot 0x00000004
+#endif
+
+#ifndef kFSEventStreamCreateFlagIgnoreSelf
+#define kFSEventStreamCreateFlagIgnoreSelf 0x00000008
+#endif
+
+#ifndef kFSEventStreamCreateFlagFileEvents
+#define kFSEventStreamCreateFlagFileEvents 0x00000010
+#endif
+
+static napi_value FSEStart(napi_env env, napi_callback_info info) {
+  size_t argc = 2;
+  napi_value argv[2];
+  char path[PATH_MAX];
+
+  assert(napi_get_cb_info(env, info, &argc, argv,  NULL, NULL) == napi_ok);
```

Comment:
Note that wrapping everything in asserts means a NDEBUG build will be completely broken. I don't know how common that is but you might want to copy core's CHECK macro:
```
#include <assert.h.
#include <stdlib.h>

#ifdef NDEBUG
#define CHECK(x) do { if (!(x)) abort(); } while (0)
#else
#define CHECK assert
#endif
```